[GTK] Improve testing setup for aarch64 platforms

philn opened the pull request:
#### 16da99fe7d1b071f64f4ac4d9900250aee4cc4bd
<pre>
[GTK] Improve testing setup for aarch64 platforms
<a href="https://bugs.webkit.org/show_bug.cgi?id=253166">https://bugs.webkit.org/show_bug.cgi?id=253166</a>

Reviewed by Carlos Alberto Lopez Perez.

Do not hardcode the host machine information when building the llvmpipe path.

* Tools/Scripts/webkitpy/port/gtk.py:
(GtkPort.setup_environ_for_server):

Canonical link: <a href="https://commits.webkit.org/261062@main">https://commits.webkit.org/261062@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/95a69936be3db9dfeb153606ccaa48e9ce6f0015

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/110491 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/19575 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/43081 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/1860 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/119379 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/114438 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/21005 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/10723 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/102726 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/116233 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/15639 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/98823 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/43872 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/113910 "Passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/97601 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/30494 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/85739 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/12229 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/31829 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/12806 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/8795 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/18169 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/51449 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7660 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/14673 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d4de672446d396b2a9b16117807a2b92475c6d4a. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/10876

clopez commented on Tools/Scripts/webkitpy/port/gtk.py:
```diff
@@ -81,7 +82,7 @@ def setup_environ_for_server(self, server_name=None):
                                                                     ignore_errors=True).strip()
                 dri_libgl_path = os.path.join(llvmpipe_libgl_path, "dri")
             else:  # in flatpak
-                llvmpipe_libgl_path = "/usr/lib/x86_64-linux-gnu/"
+                llvmpipe_libgl_path = "/usr/lib/{}-linux-gnu/".format(platform.machine())
```
You can use also `os.uname().machine` and avoid the import of the platform module.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/95a69936be3db9dfeb153606ccaa48e9ce6f0015. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/10876

philn committed:
[GTK] Improve testing setup for aarch64 platforms
https://bugs.webkit.org/show_bug.cgi?id=253166

Reviewed by Carlos Alberto Lopez Perez.

Do not hardcode the host machine information when building the llvmpipe path.

* Tools/Scripts/webkitpy/port/gtk.py:
(GtkPort.setup_environ_for_server):

Canonical link: https://commits.webkit.org/261062@main

webkit-commit-queue commented:
Committed 261062@main (16da99fe7d1b): <https://commits.webkit.org/261062@main>

Reviewed commits have been landed. Closing PR #10876 and removing active labels.
